ui-core/src/components/inputs/InputWithSuggestions/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  ChangeEventHandler,
+  KeyboardEventHandler,
+  ReactNode,
+  FocusEventHandler,
+} from 'react';
+import cx from 'classnames';
+import Input, { InputProps } from '../Input';
+import { normalizeString } from './utils';
+
+interface InputWithSuggestionsProps<T> extends Omit<InputProps, 'onChange'> {
+  suggestions: Array<T>;
+  onChange: (value: T) => void;
+  getSuggestionLabel: (option: T) => string;
+  customLabel?: ReactNode;
+  numberOfSuggestionsToShow?: number;
+  exactSearch?: boolean;
+  defaultValue?: string;
+  withChevron?: boolean;
+}
+
+const InputWithSuggestions = <T,>({
+  suggestions,
+  onChange,
+  getSuggestionLabel,
+  customLabel,
+  numberOfSuggestionsToShow = 5,
+  exactSearch = false,
+  defaultValue = '',
+  withChevron,
+  ...inputProps
+}: InputWithSuggestionsProps<T>) => {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestion, setActiveSuggestion] = useState(-1);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [suggestionClicked, setSuggestionClicked] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.value = defaultValue;
+    }
+  }, [defaultValue]);
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const userInput = normalizeString(e.currentTarget.value);
+
+    if (userInput === '') {
+      setFilteredSuggestions([]);
+      setShowSuggestions(false);
```

Comment:
It would be nice if we could use a simple variable for `showSuggestions` instead of a state variable.

Reply:
Nice hint, fixed